src/overlays/actors/ovl_Dm_Char07/z_dm_char07.c

```diff
@@ -21,16 +22,157 @@ const ActorInit Dm_Char07_InitVars = {
     (ActorFunc)DmChar07_Update,
     (ActorFunc)DmChar07_Draw,
 };
-*/
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Dm_Char07_0x80AAE9C0/func_80AAE9C0.asm")
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Dm_Char07_0x80AAE9C0/DmChar07_Init.asm")
+extern Gfx D_06000100[];
+
+extern Gfx D_06000240[];
+
+extern Gfx D_06000650[];
+
+extern Gfx D_06000790[];
+
+extern Gfx D_06000B80[];
+
+extern Gfx D_06000CC0[];
+
+extern Gfx D_060010D0[];
+
+extern Gfx D_06001210[];
+
+extern Gfx D_060015E0[];
+
+extern Gfx D_060016B8[];
+
+extern Gfx D_06002BA0[];
+
+extern Gfx D_06002CD0[];
+
+extern CollisionHeader D_06006688;
+
+extern Gfx D_060076A0[];
+
+extern Gfx D_06007918[];
+
+extern AnimatedMaterial D_060105F8;
+
+extern Gfx D_06010D68[];
+
+void DmChar07_SetupAction(DmChar07 *this, DmChar07ActionFunc actionFunc) {
+    this->actionFunc = actionFunc;
+}
+
+void DmChar07_Init(Actor *thisx, GlobalContext *globalCtx) {
+    DmChar07 *this = THIS;
+
+    this->unk2BA = 0;
```

Comment:
```suggestion
    this->onStageFlag = false;
```
From looking at this in KZ and the code, it looks like this is some sort of flag when the actor/player is on stage to zoom out the camera/scale

Reply:
I don't think it is. Looking at it in KZ, i am getting a value of 1 even when I'm not on the stage.

Reply:
Looks like it a flag whether that particular instance is the stage or not. The stage is the only one that has some collision, (surprisingly it is only the flower pots in the back that have collision, there is something below it that you actually stand on), so it uses this to make sure to delete the collision on destroy.